Dockerfile

```diff
@@ -0,0 +1,7 @@
+FROM alpine:latest
+RUN apk --no-cache add curl jq && \
+    SDK_VERSION=$(curl -sL https://api.github.com/repos/snyk/snyk-iac-rules/releases/latest | jq -r ".tag_name" | cut -c2-) && \
+    wget https://github.com/snyk/snyk-iac-rules/releases/latest/download/snyk-iac-rules_${SDK_VERSION}_Linux_x86_64.tar.gz && \

```

Comment:
I know we discussed how the Docker auto builder will trigger a build of this and take care of multiple environments. Just have two questions because I don't know enough about this process:
1. What tagging does it use? We need to use semver for our security scanning I believe
2. I assume since this image is built on alpine that it's okay to download the Linux binary here. Just want to confirm

Reply:
2. -> Afaik, if the binary is built with CGO_ENABLED=0 (and if the binary is originating from [here](https://github.com/snyk/snyk-iac-rules/blob/develop/.goreleaser.yml) this is the case), it should be good. I did a quick test, _build_ & _template_ worked fine. I wanted to test 'push' ( as the 'net' package usage could be problematic) and got an unauthenticated error. @teodora-sandu I understand we use oras lib to push content to OCI registries, and oras expects a '$HOME/.docker/config.json' to be in place.  So after having configured my local docker to use a 'naive' credsstore (because by default picks a OS X credstore) and copying the docker config to the right folder in the container, pushing worked too.

We will need to document how the authentication would work for the container use case.

Reply:
Interesting, do you think it should be on us to teach our customers how to configure their docker to be authenticated? At the moment we only mention in the `--help` and in https://docs.snyk.io/products/snyk-infrastructure-as-code/custom-rules/getting-started-with-the-sdk/pushing-a-bundle that they need to authenticate

Reply:
> 

1. AFAIK the tagging is done in Docker Hub's auto builder.
2. I assumed the same thing and when I tested it locally it worked without any problem.

Reply:
> Interesting, do you think it should be on us to teach our customers how to configure their docker to be authenticated? At the moment we only mention in the `--help` and in https://docs.snyk.io/products/snyk-infrastructure-as-code/custom-rules/getting-started-with-the-sdk/pushing-a-bundle that they need to authenticate

If we provide a container image, providing some extra help with regards to how they can copy an existing docker config (that does not involve some Window & Mac OS related password helpers) would definitely help customers. There is not docker or oras client in the container mind, they can't just call 'docker login' or oras login from within the container and that's what's recommended in [here](https://docs.snyk.io/products/snyk-infrastructure-as-code/custom-rules/getting-started-with-the-sdk/pushing-a-bundle). A lot of container-savy customers can figure it own their own, just like I did. However if we're providing a container image, we might as well try to create better experience by creating enough documentation. If we don't want to invest here at the moment, that's fine too.

Reply:
To make this simpler, for containerized usage could we add a flag to the interface to specify the path to the configuration file? I'm sure we can think of an alternative approach to having to copy configuration files into the container?

Reply:
@aron even with a file flag though, the docker config file would need to be available locally in the container, no? What we could do is take in the registry credentials (url, username, password) an create a temp config file on the fly, or configure oras CLI during runtime (if possible at all) so that we don't write any pw containing files anywhere.

Reply:
I think we can pass that directly into [`NewClient()`](https://pkg.go.dev/oras.land/oras-go@v1.0.0/pkg/auth/docker#NewClient) but there's certainly other options. Passing those flags as you suggest would be a better user experience than having to manually deal with configuration files.

Reply:
We have https://snyksec.atlassian.net/browse/CFG-1192 for allowing the SDK to do logins, maybe we could move this conversation there?